zkevm-circuits/src/pi_circuit.rs

```diff
@@ -1456,13 +1456,40 @@ pub struct PiTestCircuit<F: Field, const MAX_TXS: usize, const MAX_CALLDATA: usi
 );
 
 #[cfg(any(feature = "test", test, feature = "test-circuits"))]
-impl<F: Field, const MAX_TXS: usize, const MAX_CALLDATA: usize>
-    PiTestCircuit<F, MAX_TXS, MAX_CALLDATA>
+impl<F: Field, const MAX_TXS: usize, const MAX_CALLDATA: usize> SubCircuit<F>
+    for PiTestCircuit<F, MAX_TXS, MAX_CALLDATA>
 {
+    type Config = PiCircuitConfig<F>;
+
+    fn new_from_block(block: &witness::Block<F>) -> Self {
+        let mut block = block.clone();
+        block.circuits_params.max_txs = MAX_TXS;
+        block.circuits_params.max_calldata = MAX_CALLDATA;
+
+        Self(PiCircuit::new_from_block(&block))
+    }
+
+    fn min_num_rows_block(block: &witness::Block<F>) -> (usize, usize) {
+        let mut block = block.clone();
+        block.circuits_params.max_txs = MAX_TXS;
+        block.circuits_params.max_calldata = MAX_CALLDATA;
```

Comment:
```suggestion
        assert_eq!(block.circuits_params.max_txs, MAX_TXS);
        assert_eq!(block.circuits_params.max_calldata, MAX_CALLDATA);
```
With this you could also skip the clone.